v2/event/extensions.go

```diff
@@ -48,6 +48,24 @@ func validateExtensionName(key string) error {
 		return fmt.Errorf("bad key, CloudEvents attribute name '%s' is longer than %d characters", key, MaxExtensionNameLength)
 	}
 
+	specV1Attributes := map[string]struct{}{
+		"id":              {},
+		"subject":         {},
+		"time":            {},
+		"source":          {},
+		"specversion":     {},
+		"type":            {},
+		"datacontenttype": {},
+		"dataschema":      {},
+	}
+
+	// TODO: not safe, can be bypassed with a ce-ce- prefix
+	trimmed := strings.TrimPrefix(key, "ce-")
```

Comment:
not needed, if the key has a `-` it is not a valid extension per the test on 70.